src/webfaf/problems.py

```diff
@@ -549,10 +596,18 @@ def item(problem_id):
     return render_template("problems/item.html", **forward)
 
 
-@problems.route("/bthash/", endpoint="bthash_permalink")
-@problems.route("/bthash/<bthash>/")
+@problems.route("/bthash/", endpoint="bthash_permalink", methods=["GET", "POST"])
+@problems.route("/bthash/<bthash>/", methods=["GET", "POST"])
 def bthash_forward(bthash=None):
     # single hash
+
+    # component ids can't be accessed through components_form object because of
+    # redirection, it must be passed to the item function as an parameter
+    if request.method == 'POST':
+        component_names = request.form.get('component_names')
+    else:
+        component_names = None
```

Comment:
?? Why you are ignoring the GET?

Reply:
we are not ignoring GET, but only in POST there can come `component_names`.